src/main/kotlin/org/jetbrains/teamcity/maven/sdk/AbstractTeamCityMojo.kt

```diff
@@ -170,6 +179,105 @@ public abstract class AbstractTeamCityMojo() : AbstractMojo() {
+                return false
+            } else {
+                try {
+                    password = tokenFile.readText().toLong().toString()
+                    log.debug("Using $password maintenance token to authenticate")
+                } catch (ex: NumberFormatException) {
+                    log.warn("Malformed maintenance token: should contain a number")
+                    return false
+                }
+            }
+        }
+
+        val authToken = "Basic " + String(Base64.getEncoder().encode("$username:$password".toByteArray(Charset.forName("UTF-8"))))
+
+        val url = URL("http://$serverAddress/httpAuth/admin/plugins.html?action=setEnabled&enabled=false&pluginPath=%3CTeamCity%20Data%20Directory%3E/plugins/$pluginPackageName")
+        log.debug("Sending " + url.toString() + "...")
+        val disableRequest = url.openConnection()
+        (disableRequest as HttpURLConnection).requestMethod = "POST"
+        disableRequest.setRequestProperty ("Authorization", authToken)
+
+        try {
+            disableRequest.getInputStream().use {
+                val result = it.buffered().bufferedReader(Charset.defaultCharset()).readLine()
+                if (!result.contains("Plugin unloaded successfully")) {
+                    if (result.contains("Plugin unloaded partially")) {
+                        log.warn("Plugin unloaded partially - some parts could still be running. Server restart could be needed. Check all resources are released and threads are stopped on server shutdown. ")
+                    } else {
+                        log.warn(result)
+                        return false
+                    }
+                } else {
+                    log.info("Plugin successfully unloaded")
+                }
+            }
+        } catch (ex: ConnectException) {
+            log.warn("Cannot find running server on http://$serverAddress. Is server started?")
+            return false;
+        } catch (ex: IOException) {
+            when (disableRequest.responseCode) {
+                401 -> log.warn("Cannot authenticate server on http://$serverAddress with " +
+                        if (username.isEmpty())
+                            "maintenance token $password. Check that server has already started with '-Dteamcity.superUser.token.saveToFile=true' parameter or provide username and password."
+                        else
+                            "provided credentials.")
+            }
+            log.warn("Cannot connect to the server on http://$serverAddress: ${disableRequest.responseCode}", ex)
+            return false
+        }
+
+        uploadPluginAgentZip()
+
+        val urlDisable = URL("http://$serverAddress/httpAuth/admin/plugins.html?action=setEnabled&enabled=true&pluginPath=%3CTeamCity%20Data%20Directory%3E/plugins/$pluginPackageName")
```

Comment:
it seems like name `urlDisable` could be improved for an enabling request
```suggestion
        val urlEnable = URL("http://$serverAddress/httpAuth/admin/plugins.html?action=setEnabled&enabled=true&pluginPath=%3CTeamCity%20Data%20Directory%3E/plugins/$pluginPackageName")
```

Reply:
Done